### DISCUSSION TITLE
Can we incrementally "add" data to existing series results?

### DISCUSSION THREAD

**Discussion by arclabch** [DaveSkender/Stock.Indicators#396] [Q&A] [answered]
I'm looking for a way to update charts as new data become available, without recalculating everything each time. Is there a way to add new data to an already calculated series?

**Comment by DaveSkender** [comment, marked as answer]
Currently, no.

I have been thinking about this a bit for v2 in the roadmap (#218) and do see users occasionally trying to do one point at a time (#216).  I think it makes sense in some use cases to do incremental calculations, especially for streaming models where you're dealing with real-time high frequency data.  With that said, not all indicators can support it due to the way they are calculated, but I think many do.

In any case, it gets tricky because current period values are usually based on one or more previous values, so we'll have to have more in-memory storage of intermediate calculations for persistence and more memory transactions to make this work.

For now, the good news is that this library is [quite fast](https://daveskender.github.io/Stock.Indicators/performance/), especially if you're using newer frameworks, and recalculating the whole time series is still feasible if you are not hitting memory or compute constraints.

**Reply by arclabch** [reply]
Thank you for the explanation. Appreciate it.

**Comment by DaveSkender** [comment]
Follow or continue this conversation in our [discussion on streaming indicators](https://github.com/DaveSkender/Stock.Indicators/discussions/1018) for progress and information about pre-release package versions to support live and incremental quote sources.
